pkg/rc/farm.go

```diff
@@ -130,6 +130,14 @@ START_LOOP:
 					continue
 				}
 
+				// Don't try to work on an RC that is already owned. While the LockedForOwnership flag may be stale,
+				// the nature of this function is that we (or another farm) will come back to it and the lock will be
+				// grabbed. Shortening the length of time it takes to process a list of RCs is paramount.
+				if rcField.LockedForOwnership {
```

Comment:
It sort of makes logical sense to have this check after the `shouldWorkOn` checks, but since those do label queries it makes performance sense to do it beforehand.